i-VRESSE/bartender
# Arq scheduler

Use https://arq-docs.helpmanual.io/ to create a scheduler that uses Redis as a queue and multiple workers (`bartender perform` cli).

TODO:
* [x] ArqScheduler
* [x] Build from config.yaml
* [x] Document scheduler in README
* [x] Development documentation: start redis Docker container + start worker
* [x] `bartender perform`  subcommand for worker
* ~~[ ] Add redis + workers to docker-compose~~ postponed to https://github.com/i-VRESSE/bartender/issues/30

To test:
1. Create config file with uncommented copy of https://github.com/i-VRESSE/bartender/blob/arq/config-example.yaml#L53-L64 as the first destination
2. Start Redis container with `docker run --detach --publish 6379:6379 redis:7`
3. Start `bartender serve`
4. In another shell start `bartender mix`
5. Submit a job as described at https://github.com/i-VRESSE/bartender/tree/arq#word-count-example

Fixes #21 
Should be merged after #17
